dvc_ssh/__init__.py

```diff
@@ -62,13 +61,29 @@ def _prepare_credentials(self, **config):
             or DEFAULT_PORT
         )
 
-        if config.get("ask_password") and config.get("password") is None:
-            config["password"] = ask_password(
-                login_info["host"], login_info["username"], login_info["port"]
+        login_info["password"] = config.get("password", None)
+        login_info["passphrase"] = config.get("passphrase", None)
+
+        if config.get("ask_password") and login_info["password"] is None:
+            login_info["password"] = ask_password(
+                login_info["host"],
+                login_info["username"],
+                login_info["port"],
+                "password",
+            )
+        if config.get("ask_passphrase") and login_info["passphrase"] is None:
+            login_info["passphrase"] = ask_password(
+                login_info["host"],
+                login_info["username"],
+                login_info["port"],
+                "private key passphrase",
             )
 
-        login_info["password"] = config.get("password")
-        login_info["passphrase"] = config.get("password")
+        if (
+            config.get("password_as_passphrase", True)
+            and login_info["passphrase"] is None
+        ):
+            login_info["passphrase"] = login_info["password"]

```

Comment:
Hm, do we really need this option?

Reply:
It was the cleanest way I found to don't break users relying on old behavior while still allowing users to opt-out (i.e. for fixing https://github.com/iterative/dvc/issues/8137). 

Felt better than alternatives:

- Trying to come up with conditions were misusing the asyncssh API might work.
Doesn't seem scalable or right. New edge cases might show in the future.

- Completely dropping old behavior.
Would be a breaking change for users relying on it.

Reply:
Got it, thanks for elaborating! Indeed, all options feel like a bit of a compromise. Considering that providing password/passphrase in the config is a bad practice and we even considered removing support for it, we can probably just drop old behaviour for the sake of correctness.

Though, another thing to note is that `ask_*` options are a bit of a legacy thing coming from `paramiko's` limitations, and `asyncssh` might be raising proper exceptions that would allow us to distinguish when we need to ask password or a passphrase. If you could confirm that, maybe we can just make ask_password and password obsolete and not add new options at all.